### ISSUE TITLE
bug: unique-enum-value-names throws "Duplicate "graphql" modules exception

### ISSUE THREAD

**Issue by dnalborczyk** [dotansimha/graphql-eslint#284]
`graphql-eslint`: **v0.7.2**

I checked the `package-lock`, there's only one version of **graphql** referenced: **v15.5.0** (latest)

```json
files: ['*.graphql'],
parser: '@graphql-eslint/eslint-plugin',
parserOptions: {
  schema: './schema/**/*.graphql',
},
plugins: ['@graphql-eslint'],
rules: {
  '@graphql-eslint/unique-enum-value-names': 'error',
},
```

```shell
/Users/daniel/graphql-schema/schema/StatementState.graphql
  0:1  error  Cannot use GraphQLEnumType "StatementState" from another module or realm.

Ensure that there is only one instance of "graphql" in the node_modules
directory. If different versions of "graphql" are the dependencies of other
relied on modules, use "resolutions" to ensure only one version is installed.

https://yarnpkg.com/en/docs/selective-version-resolutions

Duplicate "graphql" modules cannot be used at the same time since different
versions may have different capabilities and behavior. The data from one
version used in the function from another could produce confusing and
spurious results  @graphql-eslint/unique-enum-value-names
```

I tried another `graphql` validation rule. interestingly, e.g. rule `@graphql-eslint/variables-are-input-types` works without an error.

**Comment by dnalborczyk** [comment]
I think I found the problem: **graphql** is being bundled with the deployment. I assume this is not intentional?

from: **index.cjs.js**

```js
/**
 * Unique enum value names
 *
 * A GraphQL enum type is only valid if all its values are uniquely named.
 */
function UniqueEnumValueNamesRule(context) {
  var schema = context.getSchema();
  var existingTypeMap = schema ? schema.getTypeMap() : Object.create(null);
  var knownValueNames = Object.create(null);
  return {
    EnumTypeDefinition: checkValueUniqueness,
    EnumTypeExtension: checkValueUniqueness
  };

  function checkValueUniqueness(node) {
    var _node$values;

    var typeName = node.name.value;

    if (!knownValueNames[typeName]) {
      knownValueNames[typeName] = Object.create(null);
    } // istanbul ignore next (See: 'https://github.com/graphql/graphql-js/issues/2203')


    var valueNodes = (_node$values = node.values) !== null && _node$values !== void 0 ? _node$values : [];
    var valueNames = knownValueNames[typeName];

    for (var _i2 = 0; _i2 < valueNodes.length; _i2++) {
      var valueDef = valueNodes[_i2];
      var valueName = valueDef.name.value;
      var existingType = existingTypeMap[typeName];

      if (isEnumType(existingType) && existingType.getValue(valueName)) {
        context.reportError(new GraphQLError("Enum value \"".concat(typeName, ".").concat(valueName, "\" already exists in the schema. It cannot also be defined in this type extension."), valueDef.name));
      } else if (valueNames[valueName]) {
        context.reportError(new GraphQLError("Enum value \"".concat(typeName, ".").concat(valueName, "\" can only be defined once."), [valueNames[valueName], valueDef.name]));
      } else {
        valueNames[valueName] = valueDef.name;
      }
    }

    return false;
  }
}
```

**Comment by dnalborczyk** [comment]
bob-the-bundler/rollup probably needs to be fixed/adjusted to exclude direct module imports:

https://github.com/dotansimha/graphql-eslint/blob/e26a5d42bc9926f4e398a4569efcd0fe99cb2677/packages/plugin/src/rules/graphql-js-validation.ts#L28

**Comment by sparebytes** [comment]
I also have this problem, and yes it appears that `graphql` is being bundled into this package's distribution. Not sure why more people aren't reporting this problem.

**Comment by dotansimha** [comment]
You are totally right :) Working on a fix.

**Comment by dotansimha** [comment]
Fixed in https://github.com/dotansimha/graphql-eslint/commit/2f6313b5b712715e3516c85c9a5d97e4719795d6

**Comment by dotansimha** [comment]
Fixed in `@graphql-eslint/eslint-plugin@0.8.0`
